sdks/java/io/amazon-web-services/src/main/java/org/apache/beam/sdk/io/aws/s3/S3FileSystem.java

```diff
@@ -120,6 +124,10 @@ class S3FileSystem extends FileSystem<S3ResourceId> {
         MoreExecutors.listeningDecorator(
             Executors.newFixedThreadPool(
                 config.getS3ThreadPoolSize(), new ThreadFactoryBuilder().setDaemon(true).build()));
+
+    LOG.warn(
```

Comment:
I'd suggest to add similar warning log message to all other deprecated IO `expand` or read/write methods for more visibility of this change.

Reply:
The reason I added it here is that filesystems are loaded automatically... for all other IOs users should see a warning in their IDE / when compiling right? That feels a lot more noticeable than the log messages.

BTW, here's the reason the logs for S3FileSystem will be extremely verbose :/
Looks like there's a severe problem regarding initialisation of file systems with respect to `SerializablePipelineOptions`, though didn't get any feedback 
https://lists.apache.org/thread/sp9g643lfyj6fzg9m2pym23kjv6zkbmq

Reply:
✔️ done